### PR TITLE
[KIWI-1154]- F2F | GA4 | OPL Data Layer Push

### PR DESCRIPTION
**Pull request by GDainton**
## Proposed changes

Implemented GA4 analytics functionality in the form of a page view event in the data layer

### What changed

GA4 environment variable added. When set to true, GA4 data layer functionality used in addition to existing Universal Analytics. Page view event added into GA4 data layer.

### Why did it change

GOV.UK migrating to GA4 from UA

### Issue tracking

- [KIWI-1154](https://govukverify.atlassian.net/browse/KIWI-1154)

## Checklists

- [ ] Documented in the README
- [ ] Added screenshots to show the implementation is working
- [ ] Ran cfn-lint on any SAM templates

Screenshot 2023-10-02 at 15 37 43

Screenshot 2023-10-02 at 15 36 23

[KIWI-1154]: https://govukverify.atlassian.net/browse/KIWI-1154?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/assets/javascript/analytics/ga4/init.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/* global window */
+
+window.DI = window.DI || {}
+window.DI.analyticsGa4 = window.DI.analyticsGa4 || {};
+
+(function (analytics) {
+
+  'use strict'
+
+  // container is an optional parameter used for unit tests
```
Comment (Caitlin-cooling) [inline src/assets/javascript/analytics/ga4/init.js:10]: What container parameter are you referring to? `window.DI.analytics.vars.ga4ContainerId`?

File: .env.sample [hunk 2]
```diff
@@ -1,4 +1,8 @@
 API_BASE_URL=http://localhost:8000
-PORT=8000
+PORT=5030
 SESSION_SECRET=123456
 NODE_ENV=local
+UNIVERSAL_ANALYTICS_GTM_CONTAINER_ID=""
+GOOGLE_ANALYTICS_4_GTM_CONTAINER_ID=""
+GA4_ENABLED="true"
+ANALYTICS_DOMAIN="localhost"
```
Comment (Stelios-CB) [inline .env.sample:8]: Can we add the container ID's here

File: src/assets/javascript/analytics/ua/init.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    const sendData = window.DI.core.sendData
+
+    sendData({
+      "gtm.allowlist": ["google"],
+      "gtm.blocklist": ["adm", "awct", "sp", "gclidw", "gcs", "opt"],
+    })
+
+    const gaDataElement = document.getElementById("gaData");
+
+    const sessionJourney = getJourneyMapping(window.location.pathname);
```
Comment (Stelios-CB) [inline src/assets/javascript/analytics/ua/init.js:21]: Small change here - removed unused code please check the PR in CIC

File: src/assets/javascript/analytics/ua/init.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      "/validDrivingLicence": generateSessionJourney("authorize", "middle"),
+      "/workingCamera": generateSessionJourney("authorize", "middle"),
+      "/readyCheck": generateSessionJourney("authorize", "middle"),
+      "/downloadApp": generateSessionJourney("authorize", "middle"),
+      "/abort": generateSessionJourney("authorize", "end"),
+      "/simpleDevice": generateSessionJourney("authorize", "end"),
+    };
+
+    return JOURNEY_DATA_LAYER_PATHS[url];
+  }
```
Comment (Stelios-CB) [inline src/assets/javascript/analytics/ua/init.js:107]: Same as above

Comment (GDainton) [reply src/assets/javascript/analytics/ua/init.js:107]: Shall I comment it out like in CIC or remove entirely?

Comment (Stelios-CB) [reply src/assets/javascript/analytics/ua/init.js:107]: Leave as in CIC for now

File: template.yaml [hunk 5]
```diff
@@ -106,35 +106,45 @@ Mappings:
...
+      GTMIDGA4: "GTM-KD86CMZ"
+      GA4ENABLED: "false"
       ANALYTICSDOMAIN: "integration.account.gov.uk"
       PROXYURL: "proxy.review-o.integration.account.gov.uk"
     production:
       APIBASEURL: "https://api.review-o.account.gov.uk"
       DNSSUFFIX: "review-o.account.gov.uk"
       SESSIONTABLENAME: "f2f-front-sessions-production"
-      GTMID: "TT5HDKV"
+      GTMIDUA: "GTM-TK92W68"
```
Comment (Caitlin-cooling) [inline template.yaml:145]: I'm surprised to see the production IDs are the same as non-prod, is this definitely correct?

File: src/views/f2f/prove-identity-post-office.html [hunk 6]
```diff
@@ -4,7 +4,6 @@
 {% set hmpoTitle = landingPage.pageTitle %}
 {% set hmpoPageKey = "landingPage" %}
 {% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
-{% from "govuk/components/header/macro.njk" import govukHeader %}
```
Comment (Caitlin-cooling) [inline src/views/f2f/prove-identity-post-office.html:7]: How come we removed this?

Comment (GDainton) [reply src/views/f2f/prove-identity-post-office.html:7]: This interacts with the cookie banner in some way and prevents it from rendering

**Review by Caitlin-cooling — APPROVED** [review]
Nice work 👍

**Pull request merged**